src/nvim/api/win_config.c

```diff
@@ -185,26 +205,71 @@ Window nvim_open_win(Buffer buffer, Boolean enter, Dict(float_config) *config, E
     return 0;
   }
 
-  FloatConfig fconfig = FLOAT_CONFIG_INIT;
+  WinConfig fconfig = WIN_CONFIG_INIT;
   if (!parse_float_config(config, &fconfig, false, true, err)) {
     return 0;
   }
-  win_T *wp = win_new_float(NULL, false, fconfig, err);
+
+  bool is_split = HAS_KEY(config, float_config, split) || HAS_KEY(config, float_config, vertical);
+
+  win_T *wp = NULL;
+  if (is_split) {
+    win_T *parent = NULL;
+    if (HAS_KEY(config, float_config, win)) {
+      parent = find_window_by_handle(fconfig.window, err);
+      if (!parent) {
+        return 0;
+      } else if (parent->w_floating) {
+        api_set_error(err, kErrorTypeException, "Cannot split a floating window");
+        return 0;
+      }
+    }
+
+    if (HAS_KEY(config, float_config, vertical) && !HAS_KEY(config, float_config, split)) {
+      if (fconfig.vertical) {
+        fconfig.split = p_spr ? kWinSplitRight : kWinSplitLeft;
+      } else {
+        fconfig.split = p_sb ? kWinSplitBelow : kWinSplitAbove;
+      }
+    }
```

Comment:
This doesn't look right. Why are 'splitright' and 'splitbelow' only respected when "vertical" is specified?

Reply:
`vertical` is an alias for "create a vertical or horizontal split in the default direction," mimicking `:split`/`:vsplit`, where `split` allows specifying a direction so its target direction shouldn't be modified by any options.

Reply:
Ah, I missed the `is_split` condition.